packages/fluentui/react-builder/src/narration/NarrationComputer.tsx

```diff
@@ -23,14 +35,186 @@ export class NarrationComputer {
+          if (isSectionElementAncestor || isSectionRoleAncestor) {
+            // Begin if 2
+            skipParent = true;
+            break;
+          } // End if 2
+          ancestor = ancestor.parentElement as IAriaElement;
+        } // End while 2
+      } // End if 1
+
+      if (!skipParent) {
+        // Begin if 1
+        const definitions = SRNC.landmarksAndGroups[platform];
+        const landmarkOrGroup = [];
+
+        // Test if the parent's role or tag name exists in the landmarks and groups definitions
+        let testName = `role=${ariaParent.role}`;
+        let definition = ariaParent.role ? definitions[testName] : null;
+        if (!definition && !ariaParent.role) {
+          // Begin if 2
+          testName = ariaParent.tagName.toLowerCase();
+          definition = definitions[testName];
+        } // End if 2
+
+        // If the "aria-label" or "aria-labelledby" attribute exists together with a landmark or group element or role, or if it is the "narrateLabelIfNoRole" exception, add its value to the narration
+        if (definition || SRNC.narrateLabelIfNoRole.includes(platform)) {
+          // Begin if 2
+          const label =
+            ariaParent.ariaLabelledByElements
+              ?.map((labElement: HTMLElement) => {
+                return labElement.textContent;
+              })
+              .join(SRNC.DESCBY_AND_LABBY_SEPARATOR) ||
+            ariaParent.ariaLabel ||
+            null;
+
+          if (label) {
+            // Begin if 3
+            landmarkOrGroup.push(label);
+          } // End if 3
+        } // End if 2
+
+        // If the definition exists and it is not an exception that the landmark or group element or role also must have the "aria-label" or "aria-labelledby" attribute to be narrated, add the definition to the narration"""
+        if (definition && !SRNC.narrateOnlyWhenLabelled[platform].includes(testName)) {
+          // Begin if 2
+          landmarkOrGroup.push(definition);
+        } // End if 2
+        landmarksAndGroups.unshift(landmarkOrGroup.join(' '));
+      } // End if 1
+      ariaParent = ariaParent.parentElement as IAriaElement;
+    } // End while 1
+    return landmarksAndGroups;
+  } // End getEnteredLandmarksAndGroups
+
+  // Computes and returns the screen reader narration for the given element using the previous element and platform.
+  async getNarration(element: IAriaElement, prevElement: IAriaElement, platform: string): Promise<string> {
     let definitionName = this.getDefinitionName(element, platform, 'stateRules');
 
     // Retrieve the computed accessible node
     const node = await (window as any).getComputedAccessibleNode(element);
 
     // Compute and store all the narration parts
+    this.computeLandmarksAndGroups(element, prevElement, platform);
     this.computeUsage(definitionName, element, platform);
     this.computeDescription(definitionName, element, platform);
     this.computeNameAndTitle(node, element);
```

Comment:
You don't have to do this, just wondering what you think of moving the logic for each of these `this.compte<>` into seperate classes that encapsulate their behaviour, so the narration computer just has to instantiate each of the sub classes to use their behaviour. 

Doesn't need to be sub classes, you can probably just extract them to utility functions, so this class stays smaller.

Reply:
After the Teams conversation, resolved to keep it as it is.